lib/blue_state_digital/signup_form.rb

```diff
@@ -0,0 +1,100 @@
+module BlueStateDigital
+  class SignupFormField < ApiDataModel
+    FIELDS = [:id, :format, :label, :description, :is_required, :is_custom_field, :cons_field_id, :create_dt]
+    attr_accessor *FIELDS
+
+    # Takes a <signup_form_field> block already processed by Nokogiri
+    def self.from_xml(xml_record)
+      SignupFormField.new(id: xml_record[:id],
+                          format: xml_record.xpath('format').text,
+                          label: xml_record.xpath('label').text,
+                          description: xml_record.xpath('description').text,
+                          is_required: xml_record.xpath('is_required').text == '1',
+                          is_custom_field: xml_record.xpath('is_custom_field').text == '1',
+                          cons_field_id: xml_record.xpath('cons_field_id').text.to_i,
+                          create_dt: xml_record.xpath('create_dt').text)
+    end
+  end
+
+  class SignupForm < ApiDataModel
+    FIELDS = [:id, :name, :slug, :public_title, :create_dt]
+    attr_accessor *FIELDS
+
+    def self.clone(options)
```

Comment:
whoaa, named parameters exist in ruby?  that's what i've been wanting this whole time!  so much clearer than an options hash.